presto-orc/src/main/java/com/facebook/presto/orc/AbstractOrcRecordReader.java

```diff
@@ -237,7 +247,17 @@ public AbstractOrcRecordReader(
 
         maxBytesPerCell = new long[streamReaders.length];
 
-        nextBatchSize = initialBatchSize;
+        if (onlyFixedWidthColumns) {
+            if (totalFixedWidth == 0) {
+                nextBatchSize = MAX_BATCH_SIZE;
+            }
+            else {
+                nextBatchSize = toIntExact(min(MAX_BATCH_SIZE, max(1, maxBlockBytes / totalFixedWidth)));
```

Comment:
Shall we add a new static helper for `protected void adjustMaxBatchSize(long averageRowBytes)` so we can reuse the code?